### DISCUSSION TITLE
How to take a single point of a weighted bézier?

### DISCUSSION THREAD

**Discussion by Wilker-uwu** [NicolasKlenert/enterpolation#27] [Q&A] [answered]
Hi!

i arrived at this library after a bunch of difficulties trying to make a function for my game, and this has pretty much all i need, but the documentation is honestly difficult to navigate and wrap my head around since i'm nowhere near as skilled at math as i wanted. copy-pasting the snippet from [`elements_with_weights`](https://docs.rs/enterpolation/latest/enterpolation/bezier/struct.BezierBuilder.html#method.elements_with_weights) didn't work right away as i expected either in an otherwise empty `main()` function i tried to test out with (elevation tokens `?` and all else considered.

fast forward, i finally figured out how to make a weighted curve and make it how i want:
```rs
use enterpolation::bezier::{Bezier, BezierError};
use enterpolation::Curve;

fn main() -> Result<(), BezierError> {
    let acceleration: f64 = 1e5;
    let (min,max,stable) = (0.0,1.0,0.5);
    
    let bez = Bezier::builder()
                    .elements_with_weights([(min,acceleration),(stable,1.0),(max,acceleration)])
                    .domain(0.0, 1.0)
                    .constant()
                    .build()?;
    println!("{:?}", bez);
    bez.take(11).for_each(|t| println!("{:?}", t));
    bez.slice(0.5..0.5).take(1).for_each(|t| println!("{:?}", t));
    Ok(())
}
```

my problem now is that i want to have a function to take a single point of this curve, but i didn't find anything with a name i could use. i tried using a language model to figure something out but it just keeps hallucinating a `.at_t(t)` function that doesn't exist, and a `slice()` and `take()` either spits out an extra number i'm not gonna use, or just `NaN` from taking a single value which is useless to me. is there any way implemented i can use to do it?

thank you for your time, and thank you for having this exist in the first place, i appreciate it <3

**Comment by Wilker-uwu** [comment, marked as answer]
wait i found it finally! https://docs.rs/enterpolation/latest/enterpolation/trait.Generator.html#tymethod.gen

```rs
fn gen(&self, input: Input) -> Self::Output
```
Method to generate the element at the given input

so the final code would be
```rs
use enterpolation::bezier::Bezier;
use enterpolation::Generator;

fn bezier_centralize(stable_at: (f32, f32), t: f32, minmax: (f32, f32), mut acceleration: f32) -> f32 {
    let (min, max) = minmax;
    acceleration = 10f32.powf(acceleration);

    Bezier::builder()
        .elements_with_weights([(min, acceleration), stable_at, (max, acceleration)])
        .domain(0.0f32, 1.0f32)
        .constant()
        .build()
        .unwrap().gen(t)
}
```
edit: here's a code that's more friendly to copy-pasting :3

**Reply by NicolasKlenert** [reply]
Happy to see that you found the solution before I could answer. I added an issue to make the function more visible.

Other than that, if you have some difficulty navigating the crate, you are always welcome to open a discussion or issue about it. That way you (hopefully) get to your answers faster and I am able to improve the documentation.

**Reply by Wilker-uwu** [reply]
thank you <3
